contentScript.js

```diff
@@ -1,37 +1,92 @@
+// Extend the getPath function for getting
+// current selector with jQuery element
 jQuery.fn.extend({
-  getPath: function() {
-    var path, node = this;
+	getPath: function() {
+		var path, node = this;
 
-    while (node.length) {
-      var realNode = node[0], name = realNode.localName;
+		while (node.length) {
+			var realNode = node[0], name = realNode.localName;
 
-      if (!name) { break; }
-      name = name.toLowerCase();
+			if (!name) { break; }
+			name = name.toLowerCase();
 
-      var parent = node.parent();
+			var parent = node.parent();
 
-      var sameTagSiblings = parent.children(name);
-      if (sameTagSiblings.length > 1) {
-        allSiblings = parent.children();
-        var index = allSiblings.index(realNode) + 1;
-        if (index > 1) {
-            name += ':nth-child(' + index + ')';
-        }
-      }
+			var sameTagSiblings = parent.children(name);
+			if (sameTagSiblings.length > 1) {
+				allSiblings = parent.children();
+				var index = allSiblings.index(realNode) + 1;
+				if (index > 1) {
+						name += ':nth-child(' + index + ')';
+				}
+			}
 
-      path = name + (path ? '>' + path : '');
-      node = parent;
-    }
+			path = name + (path ? '>' + path : '');
+			node = parent;
+		}
 
-    return path;
-  }
+		return path;
+	}
 });
 
 $(function() {
-  $('*').mouseenter(function(event) {
-    chrome.runtime.sendMessage(
-        { path: $(event.target).getPath() }
-        // /* callback */ function(response) { ... }
-    );
-  });
+
+	var currentMousePos = {
+		x: -1,
+		y: -1
+	};
+
+	$(document)
+
+	// Track the current mouse position
+	.mousemove(function (event) {
```

Comment:
Instead of using too many listeners, we should track mousemove and use the ```elementFromPoint``` function